fiftyone/utils/eval/openimages.py

```diff
@@ -0,0 +1,793 @@
+            **kwargs: keyword arguments for the backend plotting method:
+
+                -   "plotly" backend: :meth:`fiftyone.core.plots.plotly.plot_pr_curves`
+                -   "matplotlib" backend: :meth:`fiftyone.core.plots.matplotlib.plot_pr_curves`
+
+        Returns:
+            one of the following:
+
+            -   a :class:`fiftyone.core.plots.plotly.PlotlyNotebookPlot`, if
+                you are working in a notebook context and the plotly backend is
+                used
+            -   a plotly or matplotlib figure, otherwise
+
+        """
+        if not classes:
+            c_ap = [(ap, c) for c, ap in self._classwise_AP.items()]
+            classes = [c for ap, c in sorted(c_ap)[-3:]]
+
+        precisions = []
+        recall = None
+        _classes = []
+        for c in classes:
+            if c in self.recall and c in self.precision:
+                r = self.recall[c]
+                p = self.precision[c]
+                pre, rec = self._interpolate_pr(p, r)
+                precisions.append(pre)
+                if recall is None:
+                    recall = rec
+
+                _classes.append(c)
+
+        return fop.plot_pr_curves(
+            precisions, recall, _classes, backend=backend, **kwargs
+        )
+
+    def _interpolate_pr(self, precision, recall, npts=101):
+        interp_pre = copy.deepcopy(precision)
+        rec = np.linspace(0, 1, npts)
+        q = np.zeros(101)
+        for i in range(len(interp_pre) - 1, 0, -1):
+            if interp_pre[i] > interp_pre[i - 1]:
+                interp_pre[i - 1] = interp_pre[i]
+
+        inds = np.searchsorted(recall, rec, side="left")
+        try:
+            for ri, pi in enumerate(inds):
+                q[ri] = interp_pre[pi]
+        except:
+            pass
+
+        return q, rec
+
+    def mAP(self, classes=None):
+        """Computes Open Images-style mean average precision (mAP) for the specified
+        classes.
+
+        See `this page <https://storage.googleapis.com/openimages/web/evaluation.html>`_
+        for more details about Open Images-style mAP.
+
+        Args:
+            classes (None): a list of classes for which to compute mAP
+
+        Returns:
+            the mAP in ``[0, 1]``
+        """
+        if classes is not None:
+            classwise_AP = []
+            for c in classes:
+                if c in self._classwise_AP:
+                    classwise_AP.append(self._classwise_AP[c])
+        else:
+            classwise_AP = list(self._classwise_AP.values())
+
+        classwise_AP = np.array(classwise_AP)
+        classwise_AP = classwise_AP[classwise_AP > -1]
+        if classwise_AP.size == 0:
+            return -1
+
+        return np.mean(classwise_AP)
+
+    @classmethod
+    def _from_dict(cls, d, samples, **kwargs):
+        return super()._from_dict(
+            d, samples, precision=d["precision"], recall=d["recall"], **kwargs,
+        )
+
+
+_NO_MATCH_ID = ""
+_NO_MATCH_IOU = -1
```

Comment:
@ehofesmann can you use `None` instead of `-1` for non-matches? I made this change in #974 for COCO-style evaluation because it makes using the IoU slider filter in the App more intuitive (the bounds of the slider will correspond to the actually observed IoUs, not always extending to -1).